cudamapper/src/matcher.cu

```diff
@@ -180,7 +180,12 @@ namespace claragenomics {
             std::uint64_t total_anchors = 0;
             std::uint32_t largest_block_size = 0;
 
-            for (std::size_t read_id = 0; read_id < index.number_of_reads(); ++read_id) {
+            auto num_reads_to_query = query_target_division_idx;
```

Comment:
Because `query_target_division` can be set to 0 (for all-vs-all) and in this case `num_reads_to_query`  will be `!=0` but `query_target_division` (which is used later on) will.

Reply:
Makes sense